src/odemis/dataio/test/tiff_test.py

```diff
@@ -2729,6 +2732,65 @@ def test_convert_openfibsem_to_odemis_metadata(self):
             else:
                 self.assertEqual(value, data.content[0].metadata[key])
 
+PYRAMID_FILENAME = "test-pyramid.ome.tiff"
+class TestTileRead(unittest.TestCase):
+
+    def tearDown(self):
+        # clean up
+        try:
+            os.remove(PYRAMID_FILENAME)
+        except Exception:
+            pass
+
+    def test_tiled_read(self):
+        """Test the tiled read functions"""
+
+        # storing full metadata for completeness
+
+        md = {
+            model.MD_PIXEL_SIZE: (7.738095238095199e-08, 7.738095238095199e-08),
+            model.MD_POS: (0.051781573195293, -0.007091270569792),
+            model.MD_ACQ_DATE: 1727296375,
+            model.MD_ROTATION: 6.1837053071795856,
+            model.MD_SHEAR: -4.3021142204224836e-16,
+            model.MD_DIMS: "YX",
+        }
+
+        # create a dummy overview pyramid with the same problematic shape
+        das = model.DataArray(
+            numpy.zeros((8194, 7143), dtype=numpy.uint16), metadata=md
+        )
+
+        # export the dummy overview pyramid
+        export(PYRAMID_FILENAME, das, pyramid=True)
+
+        # open the pyramid and create a projection
+        data = open_acquisition(PYRAMID_FILENAME)
+        stream = data_to_static_streams(data)[0]
+        proj = RGBSpatialProjection(stream)
+        proj._getTilesFromSelectedArea()
+
+        das = stream.raw[0]
+        image_rect = [
+            0,
+            0,
+            das.shape[1] - 1,
+            das.shape[0] - 1,
+        ]  # [xmin, ymin, xmax, ymax]
+        try:
+            for z in range(0, das.maxzoom + 1):
+                zoom_rect = img.apply_zoom_on_image_coordinates(image_rect, z)
+                tidx = img.get_tile_indices(zoom_rect, das.tile_shape)
+
+                xmin, ymin, xmax, ymax = tidx
+
+                # assert reading tiles doesn't raise an error
+                for x in range(xmin, xmax):
+                    for y in range(ymin, ymax):
+                        raw_tile = stream.raw[0].getTile(x, y, z)
+                        # logging.debug(f"Tile {x}, {y}, {z} shape: {raw_tile.shape}")
```

Comment:
This tests more the `img` functions, so you could move this to util/test/img_test.py .

Reply:
This tests the actual reads from the tiled shadow tiff, which is only for DataArrayShadowPyramidalTIFF so I think it makes sense here. There are tests for the individuail utils in img_test.py